test/internal/kubernetes/utils_create.go

```diff
@@ -0,0 +1,513 @@
+		}
+	} else {
+		log.Printf("Env %v not set to true, skipping", envInstallAzilium)
+	}
+	return cns, nil
+}
+
+func loadCiliumOverlay(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet) (appsv1.DaemonSet, error) {
+	installFlag := os.Getenv(envInstallOverlay)
+	if overlayScenario, err := strconv.ParseBool(installFlag); err == nil && overlayScenario {
+		log.Printf("Env %v set to true, deploy azure-ipam and cilium-cni", envInstallOverlay)
+		cns.Spec.Template.Spec.InitContainers[0].Args = []string{"deploy", "azure-ipam", "-o", "/opt/cni/bin/azure-ipam"}
+
+		// setup the CNS cns overlay configmap
+		if err := MustSetupConfigMap(ctx, clientset, cnsOverlayConfigMapPath); err != nil {
+			return cns, errors.Wrap(err, "failed to setup cns overlay configMap")
+		}
+	} else {
+		log.Printf("Env %v not set to true, skipping", envInstallOverlay)
+	}
+	return cns, nil
+}
+
+func loadAzureCNIOverlay(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet) (appsv1.DaemonSet, error) {
+	installFlag := os.Getenv(envInstallAzureCNIOverlay)
+	if overlayScenario, err := strconv.ParseBool(installFlag); err == nil && overlayScenario {
+		log.Printf("Env %v set to true, deploy azure-cni and azure-cns", envInstallAzureCNIOverlay)
+		cns.Spec.Template.Spec.InitContainers[0].Args = []string{"deploy", "azure-vnet", "-o", "/opt/cni/bin/azure-vnet", "azure-vnet-telemetry", "-o", "/opt/cni/bin/azure-vnet-telemetry"}
+
+		// override the volumes and volume mounts
+		cns.Spec.Template.Spec.Volumes = volumesForAzureCNIOverlay()
+		cns.Spec.Template.Spec.InitContainers[0].VolumeMounts = dropgzVolumeMountsForAzureCNIOverlay()
+		cns.Spec.Template.Spec.Containers[0].VolumeMounts = cnsVolumeMountsForAzureCNIOverlay()
+
+		// set up the CNS configMap for azure cni overlay
+		if err := MustSetupConfigMap(ctx, clientset, cnsAzureCNIOverlayConfigMapPath); err != nil {
+			return cns, errors.Wrap(err, "failed to setup CNS configMap for azure cni overlay")
+		}
+	} else {
+		log.Printf("Env %v not set to true, skipping", envInstallAzureCNIOverlay)
+	}
+	return cns, nil
+}
+
+func loadDualstackOverlay(ctx context.Context, clientset *kubernetes.Clientset, cns appsv1.DaemonSet) (appsv1.DaemonSet, error) {
+	installFlag := os.Getenv(envInstallDualStackOverlay)
+	if dualStackOverlayScenario, err := strconv.ParseBool(installFlag); err == nil && dualStackOverlayScenario {
+		log.Printf("Env %v set to true, deploy azure-vnet", envInstallDualStackOverlay)
+		cns.Spec.Template.Spec.InitContainers[0].Args = []string{
+			"deploy", "azure-vnet", "-o", "/opt/cni/bin/azure-vnet",
+			"azure-vnet-telemetry", "-o", "/opt/cni/bin/azure-vnet-telemetry", "azure-vnet-ipam", "-o",
+			"/opt/cni/bin/azure-vnet-ipam", "azure-swift-overlay-dualstack.conflist", "-o", "/etc/cni/net.d/10-azure.conflist",
+		}
+		// setup the CNS swiftconfigmap
+		if err := MustSetupConfigMap(ctx, clientset, cnsSwiftConfigMapPath); err != nil {
```

Comment:
can we double confirm if cnsSwiftConfigMapPath is correct for dualstack overlay CNS configMap?

Reply:
When you added this a while back, was it the same configmap for podsubnet as dualstack?
https://github.com/Azure/azure-container-networking/blame/477200881d37ca9d59fa157cad2d7ec68ce54f98/test/integration/setup_test.go#L192C93-L192C93